File: MATLAB/ProjectEuler_03.m
%% Largest Prime Factor -- Solved
clear, close, clc, disp(mfilename), tic;

Number = 600851475143;

%% Solution 1: Use factor() Function
p = factor(Number);
n = p(end);

%% Solution 2: Use PrimeDivide()
% n = Number;
% LoopControl = true;
% 
% while LoopControl
%     p = PrimeDivide(n);
%     if p < n
%         n = n / p;
%     else
%         LoopControl = false;
%     end
% end

%% Display Solution
fprintf('The largest prime factor of the number 600851475143 is: %d\n', n); toc;

%% PrimeDivide()
function factor = PrimeDivide(n)
TerminationFlag = false;

    for i = 2:sqrt(n)+1
        if rem(n,i) == 0
            factor = i;
            TerminationFlag = true;
            break;
        end
    end
    
    if TerminationFlag == false
        factor = n;
    end    
end

File: Python/ProjectEuler_03.py
# Largest Prime Factor -- Solved
# Solution = 6857

import math

from Timer import Timer
Stopwatch = Timer()

def PrimeDivide(n):
    for i in range(2, round(math.sqrt(n))):
        if n % i == 0:
            return i

    return n

Number = 600851475143

while True:
    p = PrimeDivide(Number)

    if p < Number:
        Number //= p
    else:
        break

print('The largest prime factor of the number 600851475143 is: ', Number)
Stopwatch.stop()
